types/chai/index.d.ts

```diff
@@ -103,7 +104,8 @@ declare namespace Chai {
 
     export interface ExpectStatic {
         (val: any, message?: string): Assertion;
-        fail(actual?: any, expected?: any, message?: string, operator?: Operator): void;
+        fail(message?: string): never;
+        fail(actual?: any, expected?: any, message?: string, operator?: Operator): never;
```

Comment:
```suggestion
        fail(actual: any, expected: any, message?: string, operator?: Operator): never;
```

See https://www.chaijs.com/api/bdd/#method_fail